embedded/store/immustore.go

```diff
@@ -2571,13 +2571,17 @@ func (s *ImmuStore) readValueAt(b []byte, off int64, hvalue [sha256.Size]byte, s
 		val, err := s.vLogCache.Get(off)
 		if err == nil {
 			// the requested value was found in the value cache
-			copy(b, val.([]byte))
+			bval := val.([]byte)
+			n := len(bval)
 
-			if !skipIntegrityCheck && hvalue != sha256.Sum256(b) {
-				return len(b), ErrCorruptedData
+			copy(b, bval)
+
+			// length is compared to cover the corner case where empty value is stored in the cache
```

Comment:
just curious, when can an empty value be stored in the cache? Because the vlogcache is being set when appendData is called. Shouldn't we put a check there too to avoid this scenario?

Reply:
empty values are not stored in the cache, it potentially could and it shouldn't cause an issue. However, due to we associate such cache to the vlog, I'm placing the cache after knowing an offset was set